Add Dockerfile for deploying the application

jekabs-karklins opened the pull request:
This commit adds a gitlab-ci, which builds daqlite binaries and publishes them as tar archive and as a docker container.
 
The Dockerfile sets up a Ubuntu 22.04 environment with necessary packages such as tightvncserver, xfce4, qtcreator, and vim-tiny. The Dockerfile_entrypoint.sh script is included to configure the VNC server and startup script. 

Tagging a commit will produce docker image with the tag e.g. "2.0" or "2.1.beta"

jekabs-karklins committed:
Add Dockerfile for deploying the application

This commit adds a Dockerfile for deploying the application. The Dockerfile sets up a Ubuntu 22.04 environment with necessary packages such as tightvncserver, xfce4, qtcreator, and vim-tiny. It also creates directories and copies application files. The Dockerfile_entrypoint.sh script is included to configure the VNC server and startup script. The container is started with the entrypoint.sh script.

tbukovics commented on Dockerfile.deploy:
```diff
@@ -0,0 +1,54 @@
+
+# Create directories
+RUN mkdir -p /ess/ecdc/daqlite/build /ess/ecdc/daqlite/configs
+
```
I would create this dummy kafka config to prevent erros if daqlite looing for it.

```suggestion
RUN echo "{}" > /ess/ecdc/daqlite/configs/kafka-config-daqlite.json

```

tbukovics commented on Dockerfile.deploy:
```diff
@@ -0,0 +1,54 @@
+
+ENV DAQLITE_HOME="/ess/ecdc/daqlite/build"
+ENV DAQLITE_CONFIG="/ess/ecdc/daqlite/configs"
+ENV DAQLITE_PRODUCTION=false
```
change it to true

```suggestion
ENV DAQLITE_PRODUCTION=true
```

tbukovics commented on Dockerfile.deploy:
```diff
@@ -0,0 +1,54 @@
+# docker build --build-arg VNC_PASSWORD=<secret> -f Dockerfile.deploy -t registry.esss.lu.se/ecdc/ess-dmsc/daqlite:<x.x> .
```
please move these docker files into a subfolder

tbukovics commented on Dockerfile.deploy:
```diff
@@ -0,0 +1,54 @@
+RUN mkdir -p /ess/ecdc/daqlite/build /ess/ecdc/daqlite/configs
+
+# Copy application files
+COPY build/bin /ess/ecdc/daqlite/build/bin
```
I think on the daqlite machine we use daqlite/bin and we do not have the "build" before.

jekabs-karklins committed:
Update Dockerfile.deploy

Co-authored-by: tbukovics <56338104+tbukovics@users.noreply.github.com>

jekabs-karklins committed:
Update Dockerfile.deploy

Co-authored-by: tbukovics <56338104+tbukovics@users.noreply.github.com>

jekabs-karklins committed:
Refactor Dockerfile.deploy and Dockerfile_entrypoint.sh for improved deployment process

jekabs-karklins commented:
@tbukovics I have added the 
x11-xserver-utils

But unfortunately I still get an error 
RandR extension missing
When trying to change resolution after the boot.

I have added an optional parameter to for the container
VNC_RESOLUTION
that we can add in the docker compose
e.g.
    environment:
      - DAQLITE_BROKER=-b kafka2:9092
      - DAQLITE_PRODUCTION=true
      - VNC_PASSWORD=<vnc_password>
      - VNC_RESOLUTION=1920x1200 # Increase resolution for better view

tbukovics committed:
remove incorrect RUN command

tbukovics reviewed (changes requested):
I corrected the entrypoint, there was an necessary RUN.

Please add a ci pipeline for gitlab do build daqlite and if it's good then create and deploy container as latest

jekabs-karklins committed:
Update Dockerfile.deploy

jekabs-karklins committed:
Rename Dockerfile.deploy to Dockerfile

jekabs-karklins committed:
Update Dockerfile

jekabs-karklins committed:
Create .gitlab-ci.yml

jekabs-karklins committed:
Update Dockerfile

jekabs-karklins committed:
Merge pull request #16 from ess-dmsc/ECDC-4352-daqlite-docker-container-updates

Ecdc 4352 daqlite docker container updates

jekabs-karklins committed:
Update .gitlab-ci.yml

jekabs-karklins commented:
@tbukovics 
I have modified Dockerfile to also do the actual build of the binaries
And added gitlab-ci that invokes the dockerfile and publishes once done.

It is quiet slow (~10 min), and could be optimized by splitting into two images with one that does not change often. but that can be done in a separate task if needed.

I have also setup the synchronization to GitLab. So once this is merged the same flow as we have for EFU should work.

tbukovics reviewed (changes requested):
This is not exactly what I expected to have, you build daqlite product when you create the docker file, and we would like to have a simplified pipeline in gitlab which:
* builds the daqlite binary on ubuntu
* if build success, then:
   * publish tar file for ubuntu
   * create and publish docker container
   
Please also do not forget the scripts and config folders from the container

jekabs-karklins committed:
Update .gitlab-ci.yml

jekabs-karklins committed:
Update .gitlab-ci.yml

jekabs-karklins committed:
Update and rename Dockerfile to Dockerfile.deploy

jekabs-karklins committed:
Update Dockerfile.deploy

jekabs-karklins commented:
We have something like this now

![image](https://github.com/user-attachments/assets/f8718ffd-0d9a-4e84-925b-46c56073503a)
